rfcs/proposed/RFC-5-server-sdk-jwt-packages.md

```diff
@@ -0,0 +1,180 @@
+# Server SDK JWT Packages
+
+Updates RFC 1 - Server SDK JWT Packages
+
+## Problem
+Many developers, both in and outside of the company, use our APIs without using a full SDK. This can be for many reasons, but authentication is a core issue that all developers using our APIs will face. While key/secret authentication is simple to implement, the ACL-based approach with JWTs provides an additional layer of complexity. There are also competing documentation differences between how JWTs should behave, and a common package can help smooth over those issues.
+
+## Duration
+16 Aug 2023 - 30 Aug 2023
+
+## Current State
+Proposal
+
+
+## Proposers
+Chris Tankersley
+
+## Detail
+JWT authentication is being used on newer APIs being deployed by Vonage as it provides a finer grain of control over authorization between clients. New applications being developed using APIs such as Messages or Conversations are increasingly being developed by outside teams such as Pre-Sales, and there is a large need for generating JWTs easily. Vonage currently offeres two solutions:
+
+* `vonage/cli`, which can generate JWTs
+* `vonage-java-jwt`, which can generate JWTs in Java
+
+These solutions do not cover the full gamut of with internal teams need, and as we can track via API usage we have a large set of customers using our APIs outside of the Java landscape. Many customers do not use our SDKs as they are implementing our Developer Preview or Beta level products, which do not include SDK support. While customers could introduce our SDKs as dependencies, this introduces a large dependency for a small need. Many customers do not want this.
+
+All of our mainline SDKs already support generating JWTs in some capacity, but generally these JWTs are very wide in capability scope or run under the assumption they are used just for the Voice API. Some SDKs expose JWT generation in a fairly clean-room implementation inside the SDK code itself, while others are tightly coupled to their Vonage Client integrations.
+
+JWTs by themselves are an industry standard (https://jwt.io/) and there are libraries for creating raw JWT objects, but there is additional complexity in crafting the JWT to the expectations of Vonage. By offering a standalone JWT package per-language, we can offer customers a quick development win by making it easy to generate Vonage-compatible JWTs with our requirements, and continue to allow developers and customers the ability to not have to require a “kitchen sink” SDK for one small operation.
+
+## Proposed Solution
+### Packaging
+Vonage will provide a self contained JWT token generator for each language, that is automatically installed in the SDKs but can also be installed via each language’s package manager. The package will be stored as an open-source project on Github under `vonage/vonage-jwt-[language]` and will be maintained with the same level of maintenance we provide the full server SDKs.
```

Comment:
This bit we've already done, right? E.g. https://github.com/Vonage/vonage-jwt-ruby

Reply:
Correct. Everything outside of the new "Signature Verification" is staying the same